src/lens_odm_parser/core.clj

```diff
@@ -167,26 +167,20 @@
   (->> (conform-lc-kw x)
        (s/conform :odm/tx-type)))
 
-(defn- tag= [tag]
-  #(= tag (:tag %)))
+(defn- coerce-oid [s]
+  (coerce :odm.data-formats/oid s))
 
-(defn- check-tag [tag e]
-  (when-not (= tag (:tag e))
-    (throw
-      (ex-info
-        (format "Expected tag %s but got %s." (name tag) (name (:tag e)))
-        {:type ::validation-error :element e :expected-tag tag}))))
+(defn- coerce-integer [s]
+  (coerce :odm.xml/integer s))
 
-(defn- check-sub-tag [tag e]
-  (when-not (some (tag= tag) (:content e))
-    (throw
-      (ex-info
-        (format "Expected to find element with tag %s as child of element with tag %s."
-                (name tag) (name (:tag e)))
-        {:type ::validation-error :element e :expected-child-tag tag}))))
+(defn- coerce-float [s]
+  (coerce :odm.xml/float s))
 
 (defmulti parse*
-  "Parses XML Element. Dispatches by :tag."
+  "Parses a single XML Element. Dispatches by :tag. Returns a map with a single
```

Comment:
Parses a single XML Element. Dispatches by :tag. Should return a map with a single key-value pair which will be merged together with the maps from other parses. Multiple values with the same key are merged with `into`.